maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt

```diff
@@ -79,27 +84,35 @@ object DeviceService {
             idbProcess.destroy()
         })
 
-        val iosDevice = IdbIOSDevice(
-            ManagedChannelBuilder.forAddress("localhost", 10882)
-                .usePlaintext()
-                .build()
-        )
-        try {
-            // Try to connect to the device repeatedly
-            MaestroTimer.withTimeout(10000) {
-                try {
-                    // The idea is that view hierarchy is empty while device is still booting
-                    iosDevice
-                        .contentDescriptor()
-                        .get()
-                        ?.takeIf { nodes -> nodes.any { it.frame?.width != 0F } }
-                } catch (ignored: Exception) {
-                    // Ignore
-                    null
-                }
-            } ?: error("idb_companion did not start in time")
-        } finally {
-            iosDevice.close()
+        val channel = ManagedChannelBuilder.forAddress(idbHost, idbPort)
+            .usePlaintext()
+            .build()
+
+        IdbIOSDevice(channel).use { iosDevice ->
+            MaestroTimer.retryUntilTrue(timeoutMs = 2000) {
+                Socket(idbHost, idbPort).use { true }
+            } || error("idb_companion did not start in time")
+
+            // The first time a simulator boots up, it can
+            // take 10's of seconds to complete.
+            MaestroTimer.retryUntilTrue(timeoutMs = 60000) {
+                val process = ProcessBuilder("xcrun", "simctl", "bootstatus", device.instanceId)
```

Comment:
is is fine to call `xcrun simctl bootstatus` several times, from my perspective it should be just called once and then wait until timeout

Reply:
It'll return "Booting" as a status when the simulator is not finished booting